logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/ScheduleNewAssetMessageUseCase.kt

```diff
@@ -160,12 +161,21 @@ internal class ScheduleNewAssetMessageUseCaseImpl(
                 outGoingAssetUploadJob = scope.launch(dispatcher.io) {
                     launch {
                         messageRepository.observeMessageVisibility(message.id, conversationId).collect { visibility ->
-                            if (visibility == MessageEntity.Visibility.DELETED) {
-                                // If the message is deleted we cancel the upload
-                                outGoingAssetUploadJob?.cancel()
-                            }
+                            visibility.fold(
+                                {
+                                    outGoingAssetUploadJob?.cancel()
+                                    this.cancel()
+                                },
+                                {
+                                    if (it == MessageEntity.Visibility.DELETED) {
+                                        outGoingAssetUploadJob?.cancel()
+                                        this.cancel()
```

Comment:
why a double cancel ?

Reply:
outGoingAssetUploadJob?.cancel()

Should be enough

Reply:
i was testing something and forgot to remove the second one, my bad